cli.sh

```diff
@@ -436,16 +437,11 @@ git hooks accept [--shared] [trigger]
 }
 
 #####################################################
-# Returns the MD5 checksum of the hook file
+# Returns the SHA1 hash of the hook file
 #   passed in as the first argument.
 #####################################################
 get_hook_checksum() {
-    # get hash of the hook contents
-    if ! MD5_HASH=$(md5 -r "$1" 2>/dev/null); then
-        MD5_HASH=$(md5sum "$1" 2>/dev/null)
-    fi
-
-    echo "$MD5_HASH" | awk "{ print \$1 }"
+    git hash-object "$1" 2>/dev/null
```

Comment:
Same comment as in the other file, if this will invalidate all accepted hooks, maybe we should pull that out into its own change, this PR already has like 4-5 different changes together.

Reply:
Jeah, this will invalidate all accepted hooks, but now when I am thinking, we really probably either 
- need to warn the user that he needs run `git hooks shared update` 
- or we just do the update it in the legacy transform, if possible...
- or we migrate the folder name-> if possible

Reply:
This doesn't only affect shared hooks, but also regular hooks inside the repo, doesn't it? (or if not this one, then maybe the same change in base-template?)

Reply:
Jeah, actually yes. But is it really a problem, if the user has to trust the hooks again. I think this is feasible.
The other part needs probably rethoughts about the new clones which are not in `shared` folder. I think about it.

Reply:
the `git hash-object` is nice since we no more rely on md5 stuff :-)